### PR TITLE
Disable user override for FirstDayOfWeek globalization tests

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/corefx/issues/30074

cc @danmosemsft

### CODE REVIEW

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoFirstDayOfWeek.cs [hunk 1]
```diff
@@ -12,8 +12,8 @@ public class DateTimeFormatInfoFirstDayOfWeek
         public static IEnumerable<object[]> FirstDayOfWeek_TestData()
         {
             yield return new object[] { DateTimeFormatInfo.InvariantInfo, DayOfWeek.Sunday };
-            yield return new object[] { new CultureInfo("en-US").DateTimeFormat, DayOfWeek.Sunday };
-            yield return new object[] { new CultureInfo("fr-FR").DateTimeFormat, DayOfWeek.Monday };
+            yield return new object[] { new CultureInfo("en-US", false).DateTimeFormat, DayOfWeek.Sunday };
```
Comment (tarekgh) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoFirstDayOfWeek.cs]: This change is fine. 

I assume you have tested it with changing the machine settings to Monday as the first day of the week while you have en-US as default culture

**Comment by ViktorHofer** [comment]
@MattGal how expensive would it be to change the first day of week in some of our CI machines to cover all different globalization scenarios?

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoFirstDayOfWeek.cs (hunk 1 above)
Comment (ViktorHofer) [reply src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoFirstDayOfWeek.cs]: honestly, no. let me do that ;)

**Comment by tarekgh** [comment]
> how expensive would it be to change the first day of week in some of our CI machines to cover all different globalization scenarios?

I don't think we need to do that. I meant at least do it on your machine to validate the fix. if we need to enable this on CI, we'll need to run the whole tests and fix any failures before we apply this change.

**Comment by MattGal** [comment]
@ViktorHofer I'd avoid doing anything that involves changing "some of" the machines;  otherwise we can easily be in a situation where a test passes during a bug's checkin and fails later on a different CI build.

**Comment by ViktorHofer** [comment]
> if we need to enable this on CI, we'll need to run the whole tests and fix any failures before we apply this change.

the community member who reported this issue run the whole test suite locally and only these tests failed on his machine, so we should be safe here. I don't expect any other tests fail.

> I'd avoid doing anything that involves changing "some of" the machines

Haven't we done the same when we changed some machine's culture to better test culture changes? I believe we have French ones?

Maybe we don't really need it but at least the discussion seems worth to me.

**Comment by tarekgh** [comment]
it is not worth it the effort changing the CI machine configuration for just this test.

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoFirstDayOfWeek.cs (hunk 1 above)
Comment (ViktorHofer) [reply src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoFirstDayOfWeek.cs]: yes, the change works as expected. thanks @tarekgh for your help.

Comment (tarekgh) [reply src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoFirstDayOfWeek.cs]: Thanks @ViktorHofer for your help fixing the test

**Pull request merged**